cpp/lib/src/BSZUtil.cc

```diff
@@ -226,14 +227,37 @@ void ExtractYearVolumeIssue(const MARC::Record &record, std::string * const year
     if (field_008 != record.end())
         *year = field_008->getContents().substr(7, 4);
 
-    const auto field_936(record.findTag("936"));
-    if (field_936 == record.end())
+    const auto field_773(record.findTag("773"));
+    if (field_773 == record.end())
         return;
 
-    *volume = field_936->getFirstSubfieldWithCode('d');
-    *issue = field_936->getFirstSubfieldWithCode('e');
+    std::vector<std::string> filtered_dates;
+    for (const auto &field : record.getTagRange("773")) {
+        if (field.getIndicator1() == '1') {
+            for (const auto &subfield : field.getSubfields()) {
+                filtered_dates.emplace_back(explode(subfield.value_, ":"));
+            }
+            break;
+        }
+    }
+
+    *volume = filtered_dates[0];
+    *issue = filtered_dates[2];
 }
 
+std::string explode(const std::string &data, const std::string &delimiters) {
```

Comment:
There is no need to write a custom explode function. Please use StringUtil::Split() instead.